wiki/DEFINITION_OF_DONE_CRITERIA.md

```diff
@@ -0,0 +1,95 @@
+
+- **Design:** All algorithms should be designed with an eye on maintainability, performance, readability, and robustness.
+- **Testing:** The goal for automated testing is to increase the code coverage as much as possible in order to ease time spent fixing bugs and developing new features. A comprehensive set of tests lowers the possibility for regressions and the introduction of bugs as the code evolves between versions. This covers both correctness & performance. 
+- **Documentation:** User-facing documentation should be complete and descriptive. Developer-facing documentation should be used for constructs which are complex and/or not immediately obvious. 
+
+### C++
+
+#### Design
+
+- Existing prims are used wherever possible
+- Array inputs and outputs to algorithms are accepted on device
+- New prims created wherever there is potential for reuse across different algorithms or prims
+- User-facing API is stateless and follows the plain-old data (POD) design paradigm
+- Public API contains a C-Wrapper around the stateless API
+- Public API contains an sklearn-like stateful wrapper around the stateless API
+
+#### Testing
+
+- Prims: GTests with different inputs
+- Algorithms: End-to-end GTests with different inputs and different datasets
+
+#### Documentation
+
+- Complete and comprehensive Doxygen strings explaining the public API, restrictions, and gotchas. Any array parameters should also note whether the underlying memory is host or device. 
+
+
+### Python
+
+#### Design
+
+- Python class is as "near drop-in replacement" for sklearn (or relevant industry standard) API as possible
+- Python class is pickleable
+- APIs use `input_to_dev_array` to accept flexible inputs and check their datatypes
+
+#### Testing 
+
+- Pytests for wrapper functionality against sklearn using relevant skleaern datasets
+- Stress tests against reasonable inputs (e.g short-wide, tall-narrow)
+- Pytests for pickle capability
+- Pytests to evaluate correctness against sklearn on a variety of datasets
+- Add algorithm to benchmarks packages
+
+#### Documentation
+
+- Complete and comprehensive Pydoc strings explaining public API, restrictions, and gotchas
+- Docstrings include references to the underlying algorithm (e.g the paper or sklearn algo being implemented or a description of the algo used if nonstandard).
+
+
+## Algorithm Status
+
+Sometimes it is possible that only some of these requirements are satisfied before an algorithm gets released. We assign a status of `bronze`, `silver`, or `gold` to denote the amount of work necessary to provide the most robust, performant, and maintainable algorithm possible.
+
+### Bronze: 
+- Known design flaws affecting sustainability and/or performance require a significant effort / redesign
+- C++ checklist is complete but Python API has not been built
```

Comment:
```suggestion
- C++ API functionality is complete but Python API has not been built
```

Reply:
I'm not sure if I agree with this one. Hoping to keep this a set of guidelines that make very clear the steps to follow in order to define things like `API functionality`. 

Perhaps we could say `C++ API functionality (see [checklist](#C++))`?